### DISCUSSION TITLE
How to model DIAs

### DISCUSSION THREAD

**Discussion by Fowlercolea** [netbox-community/netbox#12833] [Q&A]
Hi everyone,

I'm trying to identify the best way to model dedicated internet access services provided to downstream customers within Netbox.

The underlying motivation is for a DIA to appear intuitively within a Tenancy screen, as we create a tenant for customers and associate the variety of services/equipment provided to their tenancy object.

L2VPNs work perfectly for EPL/EVPL services, intuitively attaching the 'interface' via terminations to the tenancy screen. This allows us to identify all ports associated with a customer's L2VPN from the tenancy page, without needing to dive devices/search.

For DIAs, we can assign a tenancy to the IP Address, and associate larger prefixes provided to customers in much the same way. Where it falls off is documenting/tracking the port(s) on our equipment dedicated to a customer's internet service, as we'd like quick reference to that during impacts to streamline troubleshooting.

Is anyone using Netbox to accomplish this today? If so, would you share your model/workflow for me?

I've toyed with adding a custom field to interfaces that would associate a 'Tenant' object to the physical interface, but there doesn't seem to be a way to have that present within the Tenancy Stats page.

**Comment by candlerb** [comment]
My inclination - and I haven't started to work on this yet - is to create a plugin "Service" model(*), where services can be associated to a range of objects in Netbox (e.g. physical interfaces, VLAN subinterfaces on an NNI, L2VPNs etc).  Services could then in turn be associated to Tenants (as direct customers or resellers).

Then a click on a port can quickly identify all the services which depend on that port. Equally, a click on a service will tell you all the entities which are involved in provisioning that service.

(*) I'm not sure "Service" is a good name for this, as it conflicts with the concept of TCP/UDP Services under Device/VM, but I can't think of a better one yet.

**Reply by Fowlercolea** [reply]
I think that would work for my use-case as well. Attaching tenancy to the port is a bit of a shortcut for my specific use-case, but a plugin that attaches objects to an umbrella 'service' would be much more flexible.

Service is what I would choose too, but I see the conflict potentials. Nothing else really fits the umbrella-style bucket in my vocabulary, but perhaps something like a prepended 'Tenant' could keep it distinct.

**Reply by candlerb** [reply]
I guess "RFS" for Resource-Facing Service - an [OSS](https://passionateaboutoss.com/differences-between-cfs-and-rfs/) type of term.
